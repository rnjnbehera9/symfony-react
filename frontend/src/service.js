const API_URL_Movies = 'http://localhost:8000/movies';
const API_URL_User = 'http://localhost:8000/users';

const getMovies = () => fetch(API_URL_Movies).then(response=>response.json());

const createMovies = (data) => fetch(API_URL_Movies,{
    method:'POST',
    headers: {
        'Content-Type': 'application/json',
      },
    body: JSON.stringify(data),
}).then(response => response.json());
const updateMoviesCount = (id) => fetch(`${API_URL_Movies}/${id}/count`,{
    method:'PUT',
    headers: {
        'Content-Type': 'application/json',
      }
}).then(response => response.json());
const deleteMovie = (id) => fetch(`${API_URL_Movies}/${id}`,{
    method:'DELETE',
    headers: {
        'Content-Type': 'application/json',
      }
}).then(response => response.json());
const getUsers = (data) => fetch(API_URL_User,{
    method:'POST',
    headers: {
        'Content-Type': 'application/json',
      },
    body: JSON.stringify(data),
}).then(response => response.json());

export { getMovies, createMovies, updateMoviesCount, deleteMovie, getUsers};
