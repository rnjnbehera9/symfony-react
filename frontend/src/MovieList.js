import React, { useEffect, useState } from 'react';
import { Header, Message, Table, Button } from 'semantic-ui-react';
import { getMovies, deleteMovie } from './service';
import MovieForm from './MovieForm';
import IncreaseCountButton from './IncreaseCountButton';
import { confirmAlert } from 'react-confirm-alert'; // Import
import 'react-confirm-alert/src/react-confirm-alert.css'; // Import css
import {useNavigate} from 'react-router-dom';


const MovieList = () => {
    const navigate = useNavigate(); 
    const userInfo = localStorage.getItem("userinfo");
    const navigateToLogin = () => {
     if(!userInfo){
       navigate('/');
     }
     
    };
    const [movies, setMovies] = useState([]);
    const [deleteMsg, setDeleteMsg] = useState('');
    useEffect(() => {
      navigateToLogin();
      fetchMovies();
    }, []);
  
    const fetchMovies = async () => {
        const response = await getMovies();
        setMovies(response.data);
        setDeleteMsg('')
    };
  
    const handleDelete = async (id) => {
        const response = await deleteMovie(id);
        setDeleteMsg(response.data[0].message)
        fetchMovies();
    };

    const handleOnClick = async (id) => {
        confirmAlert({
            title: 'Confirm to submit',
            message: 'Are you sure to do this.',
            buttons: [
              {
                label: 'Yes',
                onClick: () => handleDelete(id)
              },
              {
                label: 'No',
                onClick: () =>  fetchMovies()
              }
            ]
          });
      }
    const onIncrease = async (data) =>{
        if(data){
            setMovies(data);
        }
        setDeleteMsg('')
    }
    const onAddition = async (data) =>{
        if(data){
            setMovies(data);
        }
        setDeleteMsg('')
    }
    return (
      <div className="App">
        <header className="App-header">
        <Header as="h1">My Movies</Header>
              {movies &&
                  <div>
                { 
                deleteMsg &&
                    <Message
                        error
                        content={deleteMsg}
                    />
                    }
                      <table>
                          <thead>
                              <tr>
                                  <th>ID</th>
                                  <th>Title</th>
                                  <th>Movie Count</th>
                                  <th>Actions</th>
                                  <th></th>
                              </tr>
                          </thead>
                          <tbody>
                          {movies.map(movie => (
  
                                      <tr id={movie.id} key={movie.id}>
                                          <td>{movie.id}</td>
                                          <td>{movie.title}</td>
                                          <td>{movie.count}</td>
                                          <td>
                                            <IncreaseCountButton onIncrease={onIncrease} movieId={movie.id} /> 
                                          </td>
                                          <td>
                                             <Button onClick={() => handleOnClick(movie.id)}>Delete</Button>  
                                          </td>
                                      </tr>
                          ))}
                          </tbody>
                      </table>
                      <MovieForm onAddition={onAddition} />
                  </div>
              }
  
        </header>
      </div>
    );
}

export default MovieList