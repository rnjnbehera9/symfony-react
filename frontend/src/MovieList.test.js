import React from 'react';
import { render, screen, waitFor } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import MovieList from './MovieList';
import { getMovies } from './service';

jest.mock('./service');

test('renders movies list', async () => {
    getMovies.mockResolvedValue([{ id: 1, title: 'Test Movie', count: 1 }] );

    render(
        <BrowserRouter>
            <MovieList />
        </BrowserRouter>
    );

    await waitFor(() => screen.getByText('Test Movie'));

    expect(screen.getByText('Test Movie')).toBeInTheDocument();
});

it('renders correctly', () => {
    const tree = renderer
      .create(        
      <BrowserRouter>
        <MovieList />
    </BrowserRouter>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });