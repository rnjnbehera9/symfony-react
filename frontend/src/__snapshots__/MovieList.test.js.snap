// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <h1
      className="ui header"
    >
      My Movies
    </h1>
    <div>
      <table>
        <thead>
          <tr>
            <th>
              ID
            </th>
            <th>
              Title
            </th>
            <th>
              Movie Count
            </th>
            <th>
              Actions
            </th>
            <th />
          </tr>
        </thead>
        <tbody />
      </table>
      <div>
        <form
          className="ui form"
          onSubmit={[Function]}
        >
          <br />
          <div
            className="field"
          >
            <label>
              Title: 
            </label>
            <input
              onChange={[Function]}
              placeholder="enter movie title"
              required={true}
              value={Array []}
            />
          </div>
          <br />
          <button
            type="submit"
          >
            Add Movie
          </button>
        </form>
      </div>
    </div>
  </header>
</div>
`;
